build-logic/basics/src/main/kotlin/gradlebuild/basics/BuildEnvironment.kt

```diff
@@ -97,6 +97,11 @@ object BuildEnvironment {
     val isTravis = "TRAVIS" in System.getenv()
     val isJenkins = "JENKINS_HOME" in System.getenv()
     val isGhActions = "GITHUB_ACTIONS" in System.getenv()
+    val isCodeQl: Boolean by lazy {
```

Comment:
You can simply add a system property to your build to add a tag. If you check all environment variables here, I suppose all of them will be an input to the build logic, so I'd prefer not to do that.

So only add `-Dscan.tag.CODEQL` to your build invocation in the gh actions file.

Reply:
I'd prefer to tie it to something that we know that CodeQL will add to the environment, over expecting a developer to add the flag. This solution is more portable: if the CodeQL extraction stuff later gets moved to TeamCity, there won't be a need to add this flag. Additionally, if someone is testing out CodeQL locally, this will also automatically add the flag in that context as well.

This is also the solution leveraged in the GE build as well to tag their build scans.

Reply:
I am fine with detecting this from the environment.  Then please choose (or a few) environment variables, but don't read all of them and then filter.

Reply:
Okay, I've updated it to do this. This looks at a selection of 7 of the 21 `CODEQL_` environment variables injected.